umpost/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,201 @@
+import re
+import f90nml
+import warnings
+import argparse
+import errno
+from pathlib import Path
+
+
+# TODO: um2netcdf will update the way arguments are fed to `process`.
+# https://github.com/ACCESS-NRI/um2nc-standalone/issues/17
+# Update the below arguments once the changes are added.
+
+# Named tuple to hold the argument list
+ARG_NAMES = collections.namedtuple(
+    "Args",
+    "nckind compression simple nomask hcrit verbose include_list exclude_list nohist use64bit",
+)
+# TODO: Confirm with Martin the below arguments are appropriate defaults.
+ARG_VALS = ARG_NAMES(3, 4, True, False, 0.5, False, None, None, False, False)
+
+# TODO: um2nc standalone will raise more specific exceptions.
+# See https://github.com/ACCESS-NRI/um2nc-standalone/issues/18
+# Improve exception handling here once those changes have been made.
+ALLOWED_UM2NC_EXCEPTION_MESSAGES = {
+    "TIMESERIES_ERROR": "Variable can not be processed",
+}
+
+
+def set_esm1p5_fields_file_pattern(run_id):
+    """
+    Generate regex pattern for finding current experiment's UM outputs.
+
+    Parameters
+    ----------
+    run_id : 5 character run ID for the current UM simulation.
+
+    Returns
+    -------
+    fields_file_name_pattern: Regex pattern for matching fields file names
+    """
+
+    # For ESM1.5 simulations, files start with run_id + 'a' (atmosphere) +
+    # '.' (absolute standard time convention) + 'p' (pp file).
+    # See get_name.F90 in the UM7.3 source code for details.
+    fields_file_name_pattern = rf"^{run_id}a.p[a-z0-9]+$"
+
+    return fields_file_name_pattern
+
+
+def set_nc_write_path(fields_file_path, nc_write_dir):
+    """
+    Set filepath for writing NetCDF to based on fields file name.
+
+    Parameters
+    ----------
+    fields_file_path : path to single UM fields file to be converted.
+    nc_write_dir : path to target directory for writing NetCDF files.
+
+    Returns
+    -------
+    nc_write_path : path for writing converted fields_file_path file.
+    """
+    fields_file_path = (
+        Path(fields_file_path)
+        if isinstance(fields_file_path, str)
+        else fields_file_path
+    )
+
+    nc_write_dir = Path(nc_write_dir) if isinstance(nc_write_dir, str) else nc_write_dir
+
+    fields_file_name = fields_file_path.name
+    nc_name = fields_file_name + ".nc"
+    nc_write_path = nc_write_dir / nc_name
+
+    return nc_write_path
+
+
+def find_matching_fields_files(dir_contents, fields_file_name_pattern):
+    """
+    Find files in list of paths with names matching fields_file_name_pattern.
+    Used to find ESM1.5 UM outputs in a simulation output directory.
+
+    Parameters
+    ----------
+    dir_contents : list of file paths, typically contents of a single directory.
+    fields_file_name_pattern : Regex pattern for matching fields file names.
+
+    Returns
+    -------
+    fields_file_paths : subset of dir_contents with names matching fields_file_name_pattern.
+    """
+
+    fields_file_paths = []
```

Comment:
This block could be pythonised into a pair of list comprehensions:

```Python
...
filepaths = [Path(filepath) for filepath in dir_contents]
fields_file_paths = [p for path in filepaths if re.match(...)]
...
```
It's more of a style thing which avoids "manual" list ops.

Reply:
Good idea, added this in